addon/-private/system/model/internal-model.js

```diff
@@ -278,8 +290,22 @@ export default class InternalModel {
     this._triggerDeferredTriggers();
   }
 
-  recordObjectWillDestroy() {
+  resetRecord() {
     this.record = null;
```

Comment:
is there actual value in reseting the record during destroy, or can we just let it go dead.